query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -68,4 +71,23 @@ public static Expression createNullNumberLiteralExpression() {
                 .setValue(Value.newBuilder().setValueType(ValueType.NULL_NUMBER)))
         .build();
   }
+
+  public static boolean isDateTimeFunction(Expression expression) {
+    return expression.getValueCase() == ValueCase.FUNCTION
+        && expression.getFunction().getFunctionName().equals("dateTimeConvert");
+  }
+
+  public static boolean isTimeColumn(String columnName) {
+    return columnName.contains("startTime") || columnName.contains("endTime");
+  }
+
+  public static boolean isComplexAttribute(Expression expression) {
+    return expression.getValueCase().equals(ATTRIBUTE_EXPRESSION)
+        && expression.getAttributeExpression().hasSubpath();
```

Comment:
Can we add this also ?
```
  private boolean isMapField(String columnName) {
    return viewDefinition.getColumnType(columnName) == ValueType.STRING_MAP;
  }
```

Reply:
didn't get your point, this method is part of the change?

Reply:
I was suggesting checking the complex attribute by checking from viewDefinition. Or this can be done while pre-processing ?

Reply:
I'd strongly recommend landing the attribute expression stuff before this - two concurrent major changes is asking for trouble.

Reply:
> I was suggesting checking the complex attribute by checking from viewDefinition. Or this can be done while pre-processing ?

The viewDefinition for prometheus is different than pinot, there is no concept of map columns here, so probably any query on map columns will not be eligible for prometheus handler (mapping column in viewDef will be missing)